shared/middleware/helpers/file_bucket.rb

```diff
@@ -13,14 +13,37 @@ def allowed_file_type?(extension)
     Sinatra::Base.mime_type(extension)
   end
 
+  def get_manifest(channel_id)
+    manifest_result = get(channel_id, FileBucket::MANIFEST_FILENAME)
+    if manifest_result[:status] == 'NOT_FOUND'
+      return []
+    end
+    JSON.load manifest_result[:body]
+  end
+
   def copy_files(src_channel, dest_channel, options={})
     # copy everything except the manifest
     options[:exclude_filenames] = [MANIFEST_FILENAME]
     result = super src_channel, dest_channel, options
+    # return right away if there are no files in the source project
+    return if result.empty?
```

Comment:
yes, I'm not quite clear on that either. I didn't write the original `copy_files` - but I'm trying to make the files API version behave like the other versions.

I think you are correct that we should not return `nil`. I'm going to update this in my next PR (coming later today). The tests look at the return value. The dashboard calls this, but doesn't look at the return value. There is one assets specific method `copy-assets` that does look at the return. Since it (plus the tests) call `to_json` on the return, I think returning an empty array makes more sense here.